test/after-throw.test.js

```diff
@@ -1,24 +1,39 @@
 'use strict'
 
-const { test } = require('tap')
+const { test } = require('node:test')
 const boot = require('..')
+const { setTimeout } = require('timers/promises')
 
-test('catched error by Promise.reject', (t) => {
+test('catched error by Promise.reject', async (t) => {
   const app = boot()
   t.plan(2)
 
-  t.threw = function (err) {
-    t.equal(err.message, 'kaboom2')
-  }
+  const uncaughtExceptionHandlers = process.listeners('uncaughtException')
 
-  app.use(function (f, opts) {
-    return Promise.reject(new Error('kaboom'))
-  }).after(function (err) {
-    t.equal(err.message, 'kaboom')
-    throw new Error('kaboom2')
-  })
+  process
+    .removeAllListeners('uncaughtException')
+    .once('uncaughtException', (err) => {
+      t.assert.strictEqual(err.message, 'kaboom2')
+
+      // Restore original handlers
+      uncaughtExceptionHandlers.forEach((handler) =>
+        process.on('uncaughtException', handler)
+      )
+    })
 
+  app
+    .use(function (f, opts) {
+      return Promise.reject(new Error('kaboom'))
+    })
+    .after(function (err) {
+      t.assert.strictEqual(err.message, 'kaboom')
+      throw new Error('kaboom2')
+    })
+
```

Comment:
what is this deviation?

Reply:
for tap the test was overriding the `threw` function, but this is not possible for node test runner.
The only idea that came into my mind is to add a single `uncaughtException` that does the check we need.

Reply:
This is SUPER strange behavior. I would not expect an uncaught promise rejection. It sounds more like a bug whch was covered by tap.